src/hotspot/share/logging/logConfiguration.cpp

```diff
@@ -564,12 +564,18 @@ void LogConfiguration::print_command_line_help(outputStream* out) {
   out->print_cr(" file=<filename>");
   out->print_cr("  If the filename contains %%p and/or %%t, they will expand to the JVM's PID and startup timestamp, respectively.");
   out->print_cr("  Additional output-options for file outputs:");
-  out->print_cr("   filesize=..  - Target byte size for log rotation (supports K/M/G suffix)."
-                                    " If set to 0, log rotation will not trigger automatically,"
-                                    " but can be performed manually (see the VM.log DCMD).");
-  out->print_cr("   filecount=.. - Number of files to keep in rotation (not counting the active file)."
-                                    " If set to 0, log rotation is disabled."
-                                    " This will cause existing log files to be overwritten.");
+  out->print_cr("   filesize=..       - Target byte size for log rotation (supports K/M/G suffix)."
+                                         " If set to 0, log rotation will not trigger automatically,"
+                                         " but can be performed manually (see the VM.log DCMD).");
+  out->print_cr("   filecount=..      - Number of files to keep in rotation (not counting the active file)."
+                                         " If set to 0, log rotation is disabled."
+                                         " This will cause existing log files to be overwritten.");
+  out->print_cr("   foldmultilines=.. - The switch to replace newline characters with multiline log event"
+                                         " with the character sequence."
+                                         " Escape newline (\\n) and backslash (\\) characters in the UL output"
+                                         " if it is set to true."
+                                         " Note that it works on file output only.");
```

Comment:
How about this message?

```
out->print_cr("   foldmultilines=.. - If set to true, a log event that consists of multiple lines"
                                       " will be folded into a single line by escaping"
                                       " the newline (\\n) and backslash (\\) characters"
                                       " in the output.")
```
I think there's no need to say "it works on file output only" because this is under the heading of "Additional output-options for file outputs".

Reply:
We are not "escaping" newline characters here. We went through great pains in the CSR request to describe the actual process accurately. We are replacing newlines with the sequence `\` and `n`. We also have to replace `\` with the sequence `\\` so that the conversion can be reversed by a parser. Saying all this in the help output is awkward but necessary. We also need to warn that this may not be safe with character encodings other than UTF-8. So something like:

`If set to true, a log event that consists of multiple lines will be folded into a single line by replacing newline characters with the sequence '\' and 'n' in the output. Existing single backslash characters will also be replaced with a sequence of two backslashes so that the conversion can be reversed. This option is safe to use with UTF-8 character encodings, but other encodings may not work.`

Reply:
Ok, this version is much better than mine :-)